### ISSUE TITLE
关于 STC 微调复现效果的疑问

### ISSUE THREAD

**Issue by czwlines** [thu-coai/CDial-GPT#33]
请问在STC微调结果中的自动评估指标BLEU-2、BLEU-4的计算方式

由于在项目中没有找到与计算BLEU相关的代码，并且在自己尝试 fine-tuning 后，发现结果与您的结果存在偏差
以下结果实在第5个epoch的生成结果，采用的是 nltk 的 corpus_bleu 工具
![image](https://user-images.githubusercontent.com/59118395/99946890-16825180-2db2-11eb-93a6-26297cf85dd6.png)

**Comment by lemon234071** [comment]
1、我们使用的NLTK bleu-2: weight[0.5 0.5], bleu-4:weight默认，smoothing method1
2、如论文中所述，我们线性递减学习率微调了10个epoch
3、您这个值比我们的要高啊，乘完100以后
4、您用的解码方法是否和我们一直呢？代码是用我们的代码吗？

**Comment by lemon234071** [comment]
此外，我们NLTK使用版本 3.4.3， 不同版本之间数值也会有问题。

**Comment by czwlines** [comment]
代码是用您的代码，模型采用的是 CDial-GPT2LCCC-base，解码部分代码是基本一致的
以下是模型 fine-tuning 的参数（_斜体是关于其他数据的参数，不产生影响_）：
Namespace(data_path='data/STC.json', dataset_cache='dataset_cache', device='cuda', eval_before_start=False, fp16='', from_step=-1, gpt2=True, gradient_accumulation_steps=64, local_rank=-1, log_file='', lr=5e-05, max_history=15, max_norm=1.0, model_checkpoint='config/cgpt2/', n_emd=768, n_epochs=70, num_workers=8, _own=False_, pretrained=True, _save_path='save/stc'_, scheduler='linear', train_batch_size=32, _train_path='data/weibo_src_tgt_utf8.train_attr.jl.prepend'_, valid_batch_size=32, _valid_path='data/weibo_src_tgt.test_attr.jl.prepend'_, valid_steps=5000, warmup_steps=5000, with_attr=False)

以下是解码的参数：
Namespace(datapath='data/STC_test.json', device='cuda', gpt2=True, max_length=100, min_length=1, model_checkpoint='save/stc11-19_16-10-21', no_sample=False, out_path='data/prediction/stc.prediction', seed=42, temperature=0.7, top_k=0, top_p=0.9)

请问测试数据是采用STC_test.json吗，其包含21161条数据

关于 NLTK 的版本，我这采用的是 3.5 版本

**Comment by lemon234071** [comment]
所以还有什么问题吗？

**Comment by czwlines** [comment]
我这样 fine-tuning 的结果有问题吗？

**Comment by lemon234071** [comment]
> 我这样 fine-tuning 的结果有问题吗？

您如果确保和我们参数一致代码一致应该是没问题的，但是您不一致，如：您微调了5个epoch
